src/clib/lib/enkf/obs_vector.cpp

```diff
@@ -81,29 +99,22 @@ __conf_instance_get_restart_nr(const conf_instance_type *conf_instance,
             conf_instance_get_item_value_double(conf_instance, "HOURS");
         util_inplace_forward_seconds_utc(&obs_time, hours * 3600);
     } else
-        util_abort("%s: Internal error. Invalid conf_instance?\n", __func__);
-
-    auto it = std::lower_bound(time_map.begin(), time_map.end(), obs_time);
-    const time_t tolerance = 30;
-    if (it == time_map.end() || std::abs(obs_time - *it) >= tolerance) {
+        throw exc::runtime_error("Observation: {} missing DATE, DAYS or HOURS",
+                                 obs_key);
+    obs_restart_nr = find_within_tolerance(time_map, obs_time);
+    if (obs_restart_nr < 0) {
+        std::string error = fmt::format(
+            "Observation: {} failed to look up report step", obs_key);
```

Comment:
Do we need the reference to `report_step` here?

Reply:
Just say: `failed to match time for..` or something instead?

Reply:
Sounds good 👍